### PR TITLE
Home page layout

### PR DESCRIPTION
**Pull request by LukeBooker**
A user can scroll through a list of pet cards, use a dropdown to filter those pets, and see a search bar at the top.

### CODE REVIEW

**Review by IanFindlay — CHANGES_REQUESTED** [review]
A lot of work particularly in terms of styling and the page looks great. I personally think that giving it another pass in order to get rid of all of the warnings - by removing the 'new' and adding some 'const' etc - and adding some commas for the sake of formatting would be worthwhile before a merge even though we will be revisiting the page in the future but let me know what you think.

File: lib/pages/home_page.dart [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+class HomePage extends StatefulWidget {
+  const HomePage({Key? key}) : super(key: key);
+
+  @override
+  State<HomePage> createState() =>
+      _HomePageState();
+}
+
+class _HomePageState extends State<HomePage> {
+  double? _deviceHeight, _deviceWidth;
```
[the remark this answers is not in the document]
Comment (LukeBooker) [reply lib/pages/home_page.dart:12]: Sounds good. I've amended.

File: lib/pages/home_page.dart [hunk 2]
```diff
@@ -0,0 +1,321 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    _deviceHeight =
+        MediaQuery.of(context).size.height;
+    _deviceWidth =
+        MediaQuery.of(context).size.width;
+    return Scaffold(
+        resizeToAvoidBottomInset: false,
+        body: Container(
```
[the remark this answers is not in the document]
Comment (LukeBooker) [reply lib/pages/home_page.dart:108]: Good spot. Think it ended up in when I was trying to create some static buttons. Removed and works fine.

File: lib/pages/home_page.dart [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+            child: Column(children: <Widget>[
+          Padding(
+            padding: const EdgeInsets.all(8.0),
+            child: TextField(
+              onChanged: (value) {},
+              controller: null,
+              decoration: InputDecoration(
```
[the remark this answers is not in the document]
Comment (LukeBooker) [reply lib/pages/home_page.dart:115]: Yes - changes made.

File: lib/pages/home_page.dart [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+                          BorderRadius.all(
+                              Radius.circular(
+                                  50.0)))),
+            ),
+          ),
+          Column(
+            children: <Widget>[
+              Container(child: filterDropDown()),
+              Container(
+                color: Color.fromARGB(
```
[the remark this answers is not in the document]
Comment (LukeBooker) [reply lib/pages/home_page.dart:134]: Cool, I've added.

File: lib/pages/home_page.dart [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+                    color: Colors.deepPurple,
+                    fontSize: 18,
+                    fontStyle: FontStyle.normal,
+                  ),
+                  prefixIcon: Icon(Icons.search),
+                  border: OutlineInputBorder(
+                      borderRadius:
+                          BorderRadius.all(
+                              Radius.circular(
+                                  50.0)))),
```
[the remark this answers is not in the document]
Comment (LukeBooker) [reply lib/pages/home_page.dart:127]: The Dart comments exist when opened with Dart in VS Code.

File: lib/pages/home_page.dart [hunk 6]
```diff
@@ -0,0 +1,321 @@
...
+              Container(
+                color: Color.fromARGB(
+                    255, 214, 201, 250),
+                height: _deviceHeight! * 0.80,
+                width: _deviceWidth!,
+                child: petCards(),
+              )
+            ],
+          )
+        ])));
```
[the remark this answers is not in the document]
Comment (LukeBooker) [reply lib/pages/home_page.dart:142]: Yes, looks much nicer now.

File: lib/pages/home_page.dart [hunk 7]
```diff
@@ -0,0 +1,321 @@
...
+        color: Colors.deepPurpleAccent,
+      ),
+      onChanged: (String? newValue) {
+        setState(() {
+          dropdownValue = newValue!;
+          filteredEntries = entries
+              .where((pet) => dropdownValue
+                  .contains(pet["species"]))
+              .toList();
+          if (filteredEntries.length == 0 &&
```
[the remark this answers is not in the document]
Comment (LukeBooker) [reply lib/pages/home_page.dart:163]: Sound good. I've amended.

[the remark this answers is not in the document]
Comment (LukeBooker) [reply lib/pages/home_page.dart:163]: Sound good. I've amended.

File: lib/pages/home_page.dart [hunk 8]
```diff
@@ -0,0 +1,321 @@
...
+              dropdownValue != "- Filter by -") {
+            showDialog(
+              context: context,
+              builder: (BuildContext context) {
+                return AlertDialog(
+                  title: new Text("Sorry!"),
```
[the remark this answers is not in the document]
Comment (LukeBooker) [reply lib/pages/home_page.dart:169]: Removed. Works fine!

File: lib/pages/home_page.dart [hunk 9]
```diff
@@ -0,0 +1,321 @@
...
+                            .pop();
+                      },
+                    ),
+                  ],
+                );
+              },
+            );
+          }
+          ;
+        });
```
[the remark this answers is not in the document]
Comment (LukeBooker) [reply lib/pages/home_page.dart:186]: It kicked up a fuss about this. I thought it looked odd. I've just managed to remove it.

File: lib/pages/home_page.dart [hunk 10]
```diff
@@ -0,0 +1,321 @@
...
+                              child: Icon(
+                                Icons.favorite,
+                                color:
+                                    Colors.pink,
+                                size: 40.0,
+                                semanticLabel:
+                                    "Heart",
+                              ),
+                            )
+                          ]))
```
[the remark this answers is not in the document]
Comment (LukeBooker) [reply lib/pages/home_page.dart]: Amended, thanks.

**Review by IanFindlay — APPROVED** [review]
Thanks for making those changes, looks good to me.

**Comment by LukeBooker** [comment]
> Thanks for making those changes, looks good to me.

**Comment by LukeBooker** [comment]
@IanFindlay Thanks Ian, really helpful comments. Spotted a few other formatting corrections as I went back through.

**Pull request merged**